src/koa-esm-specifier-transform.ts

```diff
@@ -0,0 +1,57 @@
+import getStream from 'get-stream';
+import isStream from 'is-stream';
+import * as Koa from 'koa';
+import {transformHTMLString} from './transform-html';
+import {transformJavaScriptModuleString} from './transform-javascript-module';
+
+export type TransformSpecifierFunction = (baseURL: string, specifier: string) =>
+    string;
+
+export default middleware;
+export function middleware(transformSpecifier: TransformSpecifierFunction):
+    Koa.Middleware {
+  return async (ctx: Koa.Context, next: Function) => {
+    await next()
+
+    if (ctx.response.is('html')) {
+      ctx.body = resolveSpecifiersInInlineScriptTags(
+          await getBodyAsString(ctx.body), ctx.request.url, transformSpecifier);
+    }
+
+    if (ctx.response.is('js')) {
+      ctx.body = resolveSpecifiersInJavaScriptModule(
+          await getBodyAsString(ctx.body), ctx.request.url, transformSpecifier);
+    }
+  }
+}
+
+function resolveSpecifiersInInlineScriptTags(
+    body: string,
+    requestURL: string,
+    transformSpecifier: TransformSpecifierFunction): string {
+  return transformHTMLString(body, requestURL, transformSpecifier);
+}
+
+function resolveSpecifiersInJavaScriptModule(
```

Comment:
what's the point of these two functions?

Reply:
I think I created them initially anticipating exporting them or some other behavior and that turned out to be unnecessary-- failed the YAGNI test.  Removed!